sql/src/main/java/io/crate/exceptions/Exceptions.java

```diff
@@ -76,6 +77,34 @@ public static Throwable unwrap(@Nonnull Throwable t) {
         return result;
     }
 
+    public static Throwable unwrap(@Nonnull Throwable t, Class... classes) {
+        int counter = 0;
+        Throwable result = t;
+        while (((BooleanSupplier) () -> {
```

Comment:
Could the other `unwrap` function be changed to call into this one with a static definition of classes? Or maybe change the signature to take a `Predicate<Class>` ?